package/gluon-mesh-babel/src/respondd.c

```diff
@@ -139,9 +470,39 @@ static char*  get_line_from_run(const char* command) {
 	return line;
 }
 
+__attribute__((constructor)) static void init(void) {
+	if (pthread_mutex_init(&neighbours_lock, NULL) != 0) {
+		perror("mutex init failed for neighbours");
+		exit(-2);
+	}
+
+	neighbours = json_object_new_object();
+
+	char *version = get_line_from_run("exec babeld -V 2>&1");
+	babeld_version = gluonutil_wrap_string(version);
+	free(version);
+
+	model = gluonutil_read_line("/tmp/sysinfo/model");
+
+	if (pthread_create(&babelmonitor, NULL, &babeld_monitor_thread_start, NULL) < 0 ) {
+		perror("error on pthread_create for babel monitor");
+	}
+}
+
+__attribute__((destructor)) static void deinit(void) {
+	pthread_cancel(babelmonitor);
```

Comment:
The helper thread should still be `join()`ed

Reply:
(y)